_posts/2025-01-16-wamr-2024-summary.md

```diff
@@ -0,0 +1,82 @@
+- Amazon: Contributed 14.33% of organized PRs and helped review 19.90% of PRs.
+- Xiaomi: Contributed 12.40% of organized PRs and helped review 15.11% of PRs.
+
+These contributions have been instrumental in driving WAMR forward, and we extend our heartfelt thanks to everyone involved.
+
+## New Features in WAMR 2024
+
+Several exciting new features have been added to WAMR in 2024, aimed at enhancing the development experience and expanding the capabilities of WAMR. Here are some of the key features:
+
+### Development Tools: Simplifying Wasm Development
+
+One of the most exciting additions to WAMR in 2024 is the introduction of new development tools aimed at simplifying Wasm development. These tools include:
+
+- Linux perf for Wasm Functions: This tool allows developers to profile Wasm functions directly, providing insights into performance bottlenecks.
+- AOT Debugging: Ahead-of-time (AOT) debugging support has been added, making it easier to debug Wasm applications.
+- Call Stack Dumps: Enhanced call stack dumps provide detailed information about the execution flow, aiding in troubleshooting and optimization.
+
+Before these tools, developing a Wasm application or plugin using a host language was a complex task. Mapping Wasm functions back to the source code written in the host language required deep knowledge and was often cumbersome. Debugging information from the runtime and the host language felt like two foreign languages trying to communicate without a translator. These new development tools act as that much-needed translator, bridging the gap and making Wasm development more accessible and efficient.
+
+### Shared Heap: Efficient Memory Sharing
+
+Another significant feature introduced in 2024 is the shared heap. This feature addresses the challenge of sharing memory between the host and Wasm. Traditionally, copying data at the host-Wasm border was inefficient, and existing solutions like externref lacked flexibility and toolchain support.
+
+The shared heap approach uses a pre-allocated region of linear memory as a "swap" area. Both the embedded system and Wasm can store and access shared objects here without the need for copying. However, this feature comes with its own set of challenges. Unlike memory.grow(), the new memory region isn't controlled by Wasm and may not even be aware of it. This requires runtime APIs to map the embedded-provided memory area into linear memory, making it a runtime-level solution rather than a Wasm opcode.
+
+### Newly Implemented Features
+
+Several features have been finalized in 2024, further enhancing WAMR's capabilities:
+
+- GC Features: Garbage collection features for the interpreter, LLVM-JIT, and AOT have been finalized.
+- Legacy Exception Handling: Legacy exception handling for the interpreter has been added.
+- WASI-NN: Support for WASI-NN with OpenVINO and llama.cpp backends has been introduced.
+- WASI Support: Ongoing support for WASI on ESP-IDF and Zephyr.

```

Comment:
For clarity,  this means WASI Preview 1?

Reply:
yes, I think it's WASIp1. WAMR doesn't support WASI 0.2+